parts/windows/windowscsehelper.ps1

```diff
@@ -335,4 +337,37 @@ function Install-Containerd-Based-On-Kubernetes-Version {
     $ContainerdUrl = $ContainerdUrl + $containerdPackage
   }
   Install-Containerd -ContainerdUrl $ContainerdUrl -CNIBinDir $CNIBinDir -CNIConfDir $CNIConfDir -KubeDir $KubeDir
+}
+
+function Logs-To-Events {
+    $count = $args.Count
+    $cmd = [string]$args[1..$count]
+    $cmd = $cmd.Replace('\', '\\')
+
+    $task = $args[0]
+    $eventsFileName=[DateTimeOffset]::UtcNow.ToUnixTimeMilliseconds()
+
+    $startTime=$(Get-Date -Format "yyyy-mm-dd HH:mm:ss.fff")
+    Invoke-Expression $cmd

```

Comment:
Could we log the event if it fails?

Reply:
Sure. Modify the code logic. Add log step in Set-ExiteCode function.